Sources/PrimerSDK/Classes/Core/PrimerHeadlessUniversalCheckout/Managers/VaultManager.swift

```diff
@@ -123,17 +181,46 @@ extension PrimerHeadlessUniversalCheckout {
                 
                 DispatchQueue.main.async {
                     PrimerDelegateProxy.primerDidFailWithError(err, data: self.paymentCheckoutData) { decision in
-                        // No need to pass anything
+                        // No need to do anything
                     }
                 }
                 return
             }
             
+            if let vaultedPaymentMethodAdditionalData = vaultedPaymentMethodAdditionalData {
+                if let errors = self.validateAdditionalDataSynchronously(vaultedPaymentMethodId: vaultedPaymentMethodId, vaultedPaymentMethodAdditionalData: vaultedPaymentMethodAdditionalData) {
+                    DispatchQueue.main.async {
+                        var primerError: PrimerErrorProtocol?
+                        
+                        if errors.count == 1 {
+                            if let primerErr = errors.first as? PrimerValidationError {
+                                primerError = primerErr
+                            } else if let primerErr = errors.first as? PrimerError {
+                                primerError = primerErr
+                            }
+                        }
+                        
+                        if primerError == nil {
+                            let primerErr = PrimerError.underlyingErrors(
+                                errors: errors,
+                                userInfo: nil,
+                                diagnosticsId: UUID().uuidString)
+                            primerError = primerErr
+                        }
+                        
+                        PrimerDelegateProxy.primerDidFailWithError(primerError!, data: self.paymentCheckoutData) { decision in
+                            // No need to do anything
+                        }
+                    }
+                    return
+                }
+            }
+            
             PrimerDelegateProxy.primerHeadlessUniversalCheckoutDidStartTokenization(for: vaultedPaymentMethod.paymentMethodType)
             
             let tokenizationService: TokenizationServiceProtocol = TokenizationService()
             firstly {
-                tokenizationService.exchangePaymentMethodToken(vaultedPaymentMethod.id)
+                tokenizationService.exchangePaymentMethodToken(vaultedPaymentMethod.id, vaultedPaymentMethodAdditionalData: self.vaultedPaymentMethodAdditionalData)
```

Comment:
Is `self` needed in `self.vaultedPaymentMethodAdditionalData`?

Reply:
Not really